### ISSUE TITLE
Panel Arrangement is unusable

### ISSUE THREAD

**Issue by jpryne** [Kode/KodeGarden#13]
On Windows 10, I consistently get an unusable window arrangement, as shown in the image below. This may be due to having multiple monitors, as I also have an Oculus Rift installed.

![kha](https://user-images.githubusercontent.com/1497564/55814833-27c6ff00-5aa4-11e9-89bd-894589c1be25.png)

**Comment by RobDangerous** [comment]
Different DPI scaling options per monitor? Same in other browsers?

**Comment by ianharrigan** [comment]
`Toolkit.autoScale = false` should fix that before kodegarden loads, i can make the change but need a little bit of time as i dont have anything setup for KG. Before this line should work: https://github.com/Kode/KodeGarden/blob/master/Client/src/Main.hx#L8

I would be interested to see what `Screen.instance.dpi` returns in this case also

**Comment by jpryne** [comment]
This may also be due to my Windows Accessability settings, as I often enlarge the text for easy reading.

@ianharrigan Not sure if I'm doing it right, but when I enter "Screen.instance.dpi" into the console, "TypeError: Screen.instance is undefined", though "Screen" is recognized.

**Comment by ianharrigan** [comment]
Yeah, thats probably not going to work - haxeui doesnt expose the classes by default, the `Screen` you are seeing is the browsers, not haxeui's. 

Can you run this .html (attached) click the button and paste the values? The easiest option would to probably just switch off autoscaling, or change the threshold, would be interesting to see what the values are for your system though.

Cheers,
Ian

[html5.zip](https://github.com/Kode/KodeGarden/files/3071755/html5.zip)

For reference, i get:

![image](https://user-images.githubusercontent.com/3024242/56015043-f181bd80-5cf7-11e9-8866-2b0495c50d25.png)

**Comment by jpryne** [comment]
![image](https://user-images.githubusercontent.com/1497564/56055307-0b2dff80-5d0e-11e9-934a-24e04aa9e56c.png)

**Comment by jpryne** [comment]
Also, I have a 4K monitor, though I typically have it set to 1080p.

**Comment by ianharrigan** [comment]
... that'll do it... i wonder why your dpi is so high though... Looking at how haxeui-html5 calcs this.

**Comment by ianharrigan** [comment]
```haxe
    public static function get_dpi():Float {
        if (_dpi != 0) {
            return _dpi;
        }

        var div = Browser.document.createElement("div");
        div.style.width = "1in";
        div.style.height = "1in";
        div.style.position = "absolute";
        div.style.top = "-99999px"; // position off-screen!
        div.style.left = "-99999px"; // position off-screen!
        Browser.document.body.appendChild(div);
        
        var devicePixelRatio:Null<Float> = Browser.window.devicePixelRatio;
        if (devicePixelRatio == null) {
            devicePixelRatio = 1;
        }
        
        _dpi = div.offsetWidth * devicePixelRatio;
        HtmlUtils.removeElement(div);
        return _dpi;
    }
```

Seems like its a little complex... wonder how this can be improved.

**Comment by jpryne** [comment]
Maybe it's grabbing specs from my Oculus Rift? (High resolution, small screen)

**Comment by jpryne** [comment]
My window.devicePixelRatio is 2.5 (96 * 2.5 = 240)

**Comment by ianharrigan** [comment]
i wonder why? The easy option here is to turn of auto scaling in haxeui, its not needed for KG, but its a strange one regardless

**Comment by jpryne** [comment]
I'm using a 4K uhdtv as my main screen. (I'm Batman.) :|

**Comment by ianharrigan** [comment]
is that detectable? Can you think of there being a better way to calc this dpi setting?

**Comment by jpryne** [comment]
I hope not. It's supposed to be a secr- oh, you mean the dpi? Is it important for the graphics? Seems logical to let the browser handle it.

**Comment by ianharrigan** [comment]
Yeah, autoScaling off would fix it...

**Comment by jpryne** [comment]
Ok, how do I turn off autoScaling?

**Comment by ianharrigan** [comment]
the `Toolkit.autoScale` needs to be set to false before the app starts, its not something that can be changed at runtime... i can make a "blind" PR but dont want to build it as my version of haxeui is no doubt different and i dont have a setup to run kodegarden locally at the moment.
